docs: document new Terraform cloud variables (#623)
infracost/docs: docs/features/environment_variables.md

### INFRACOST_TERRAFORM_CLOUD_HOST
For Terraform Enterprise users, used to override the default `app.terraform.io` backend host.

### INFRACOST_TERRAFORM_CLOUD_WORKSPACE

> For Terraform Enterprise users, used if you Terraform cloud workspace name is different from the Terraform workspace name.

Reviewer: ```suggestion
For Terraform Enterprise/Cloud users, used if your Terraform Cloud workspace name is different from the Terraform workspace name.
```

> Only use if Infracost cannot infer the workspace name from the Terraform cloud block configuration. If you have a valid Terraform cloud block, e.g:

Reviewer: ```suggestion
Only use if Infracost cannot infer the workspace name from the Terraform Cloud block configuration. If you have a valid Terraform Cloud block (shown below), use `INFRACOST_TERRAFORM_WORKSPACE` instead.
```

```hcl
terraform {
	cloud {

> 		organization = "infracost"

Reviewer: ```suggestion
		organization = "acmeinc"
```

		workspace {

> 			prefix = "infracost-"

Reviewer: ```suggestion
			prefix = "acme-"
```

		}
	}
}
```

> Use `INFRACOST_TERRAFORM_WORKSPACE` instead.

Reviewer: ```suggestion
```

### INFRACOST_TERRAFORM_CLOUD_ORG

> For Terraform Enterprise users, used to specify the organization name. This is only needed if you do not have a valid Terraform cloud block configuration with an organization name.

Reviewer: ```suggestion
For Terraform Cloud/Enterprise users, used to specify the organization name. This is only needed if you do not have a valid Terraform Cloud block configuration with an organization name.
```
